### PR TITLE
Bug 6903 streaming/v4

### PR DESCRIPTION
**Pull request by victorjulien**
Improve int handling. Add debug bug on's to validate that impossible conditions are in fact impossible.

### CODE REVIEW

**Review by catenacyber — APPROVED** [review]
Thanks for the work Victor

CI : 🟢 
Code : will review now
Commits segmentation : looks fine
Commit messages : Nice
Git ID set : looks fine for me
CLA : :-p
Doc update : not needed
Redmine ticket : I just set the target to 8beta1
Rustfmt : not needed
Tests : ok
Dependencies added: none added

File: src/util-decode-mime.c [hunk 1]
```diff
@@ -1093,6 +1093,8 @@ static int FindUrlStrings(const uint8_t *line, uint32_t len,
 static int ProcessDecodedDataChunk(const uint8_t *chunk, uint32_t len,
         MimeDecParseState *state)
 {
+    DEBUG_VALIDATE_BUG_ON(len > DATA_CHUNK_SIZE);
```
Comment (catenacyber) [inline src/util-decode-mime.c:1096]: Maybe add a comment that this is about 6903, and not about something mime-specific...

File: src/util-file.c [hunk 2]
```diff
@@ -662,6 +662,9 @@ static int FileStoreNoStoreCheck(File *ff)
 static int AppendData(
         const StreamingBufferConfig *sbcfg, File *file, const uint8_t *data, uint32_t data_len)
 {
+    DEBUG_VALIDATE_BUG_ON(
+            data_len > BIT_U32(26)); // 64MiB as a limit per chunk seems already excessive
```
Comment (catenacyber) [inline src/util-file.c:666]: This is good because SMB records which an be big are streamed, right ?

File: src/util-streaming-buffer.c [hunk 3]
```diff
@@ -2320,6 +2350,22 @@ static int StreamingBufferTest10(void)
...
+{
+    StreamingBufferConfig cfg = { 24, 1, STREAMING_BUFFER_REGION_GAP_DEFAULT, NULL, NULL, NULL };
+    StreamingBuffer *sb = StreamingBufferInit(&cfg);
+    FAIL_IF(sb == NULL);
+
+    StreamingBufferSegment seg1;
+    FAIL_IF(StreamingBufferAppend(sb, &cfg, &seg1, (const uint8_t *)"ABCDEFGH", 8) != 0);
+    StreamingBufferSegment seg2;
+    unsigned int data_len = 0xffffffff;
+    FAIL_IF(StreamingBufferAppend(sb, &cfg, &seg2, (const uint8_t *)"unused", data_len) != -1);
```
Comment (catenacyber) [inline src/util-streaming-buffer.c:2363]: This deserves somme comments ;-)

File: src/util-file.c (hunk 2 above)
Comment (victorjulien) [reply src/util-file.c:666]: Yes, they are streamed. So per call size should remain limited.

**Comment by victorjulien** [comment]
Merged in #10921, thanks!

**Pull request closed without merging**